Scripts/RemediationScripts/Remediate-EnableMicrosoftDefenders.ps1

```diff
@@ -97,6 +97,187 @@ function Setup-Prerequisites
+        return $_.Exception.Message
+    }
+
+    if ($currentVulnerabilitySetting -ne "MdeTvm") {
+        # Inform user and get confirmation
+        $confirmation = Read-Host "The current vulnerability setting is $currentVulnerabilitySetting. Once changed to MdeTvm, you cannot roll back to the previous setting. Do you still want to remediate? (Y/N)"
+        
+        if ($confirmation -eq "Y") {
+            $assessmentUri = "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.Security/serverVulnerabilityAssessmentsSettings/AzureServersSetting?api-version=2022-01-01-preview"
+            $assessmentBody = @{
+                kind = "AzureServersSetting"
+                properties = @{
+                    selectedProvider = "MdeTvm"
+                }
+            }
+
+            try {
+                Fetch-API -Method "PUT" -Uri $assessmentUri -Body $assessmentBody
+                Write-Host "Server vulnerability assessment settings updated to MdeTvm." -ForegroundColor $([Constants]::MessageType.Update)
+            }
+            catch {
+                Write-Host "Failed to update server vulnerability assessment settings"
+                return $_.Exception.Message
+            }
+        }
+        else {
+            Write-Host "User chose not to remediate the vulnerability setting. Skipping update."
+        }
+    }
+
+    if (-not $isWDATPEnabled) {
+        $wdAtpUri = "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.Security/settings/WDATP?api-version=2021-06-01"
+        $wdAtpBody = @{
+            id = "/subscriptions/$subscriptionId/providers/Microsoft.Security/settings/WDATP"
+            name = "WDATP"
+            type = "Microsoft.Security/settings"
+            kind = "DataExportSettings"
+            properties = @{
+                enabled = "true"
+            }
+        }
+
+        try {
+            Fetch-API -Method "PUT" -Uri $wdAtpUri -Body $wdAtpBody
+            Write-Host "WDATP setting is enabled." -ForegroundColor $([Constants]::MessageType.Update)
+        }
+        catch {
+            Write-Host "Failed to update WDATP settings"
+            return $_.Exception.Message
+        }
+    }
+
+    return $virtualMachinePricing
+}
+
+
+
+function Check-VirtualMachineCompliance {
+    param (
+        [Parameter(Mandatory=$true)]
+        [pscustomobject]$resource,
+        [Parameter(Mandatory=$true)]
+        [string]$subscriptionId,
+        [Parameter(Mandatory=$true)]
+        [string]$reqMDCTier
+    )
+
+    $isNonCompliant = $false
+    $wdAtpEnabled = $null
+    $assessmentProvider = $null
+    
+    if ($resource.PricingTier -ne $reqMDCTier) {
+        $isNonCompliant = $true
+    }
+
+    try {
+        $wdAtpResponse = Fetch-API -Method Get -Uri "https://management.azure.com/subscriptions/$subscriptionId/providers/Microsoft.Security/settings/WDATP?api-version=2022-05-01" -ErrorAction Stop
```

Comment:
NIT: Urls could be part of constants.

Reply:
Made separate variables to store these URLs